tests/components/wmspro/test_cover.py

```diff
@@ -0,0 +1,223 @@
+    mock_hub_status_prod_awning: AsyncMock,
+) -> None:
+    """Test that a cover device is created correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) == 2
+
+    device_registry = dr.async_get(hass)
+    device_entry = device_registry.async_get_device(identifiers={(DOMAIN, "58717")})
+    assert device_entry is not None
+    assert device_entry.manufacturer == MANUFACTURER
+    assert device_entry.name == "Markise"
+    assert device_entry.serial_number == 58717
+
+
+async def test_cover_update(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_awning: AsyncMock,
+) -> None:
+    """Test that a cover entity is created and updated correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) == 2
+
+    entity = hass.states.get("cover.markise")
+    assert entity is not None
+    assert entity.name == "Markise"
+
+    await async_setup_component(hass, "homeassistant", {})
+    await hass.services.async_call(
+        "homeassistant", "update_entity", {"entity_id": entity.entity_id}, blocking=True
+    )
+
+    assert len(mock_hub_status_prod_awning.mock_calls) == 3
+
+
+async def test_cover_close_and_open(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_awning: AsyncMock,
+    mock_action_call: AsyncMock,
+) -> None:
+    """Test that a cover entity is opened and closed correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) >= 1
+
+    entity = hass.states.get("cover.markise")
+    assert entity is not None
+    assert entity.state == "open"
+    assert entity.attributes["current_position"] == 100
+
+    with patch(
+        "wmspro.destination.Destination.refresh",
+        return_value=True,
+    ):
+        before = len(mock_hub_status_prod_awning.mock_calls)
+
+        await async_setup_component(hass, "cover", {})
```

Comment:
This one is not needed afaik

Reply:
Without this the service actions weren't available/found.

Reply:
I think that means something is off, they aren't needed

Reply:
If I comment out the first version of this line in `test_cover.py`, I get this:
```
FAILED tests/components/wmspro/test_cover.py::test_cover_update - homeassistant.exceptions.ServiceNotFound: service_not_found
```